lib/src/contracts/abi/abi.dart

```diff
@@ -341,12 +341,14 @@ class ContractEvent {
         // dynamic type, are not included in [topics]. A hash of the data will
         // be included instead. We can't decode these, so they will be skipped.
         final length = component.parameter.type.encodingLength;
-        if (length.isDynamic || length.length! > 32) {
+        if (length.isDynamic ||
+            length.length! > 32 ||
+            topics[topicIndex] == null) {
```

Comment:
in which case the topic can be null?

Reply:
My bad.  Apparently, this comes from an issue from `web3dart_builders` lib https://github.com/inveker/web3dart_builders/issues/10

But i haven't report in the readme these elements:

### Dart Code Generator

By using [Dart's build system](https://github.com/dart-lang/build/), webthree can
generate Dart code to easily access smart contracts.

> 
> Install web3dart_builders package
> 
> ```dart
> pub add web3dart_builders --dev
> ```

To use this feature, put a contract abi json somewhere into `lib/`.
The filename has to end with `.abi.json`.
Then, add a `dev_dependency` on the `build_runner` package and run

```
dart run build_runner build
```

You'll now find a `.g.dart` file containing code to interact with the contract.



So, like you want... it's not mandatory but if you let this, we are in the same way like web3dart

Reply:
ok, there is a case when topics can be null, thank you for merging this in :+1: